File: 프로그래머스/lv3/42579. 베스트앨범/베스트앨범.js
function solution(genres, plays) {
    const obj = {};
    const objSum = {}
    const arrSum = [];
    const result =[];
    
    genres.forEach((el,idx) => {
        obj[el] ? obj[el].push([idx,plays[idx]]) : obj[el] = [[idx,plays[idx]]];
        
        obj[el].sort((a,b) => b[1] - a[1]);
    });
    
     genres.forEach((el,idx) => objSum[el] ? objSum[el]+= plays[idx] : objSum[el] = plays[idx]);
    
    Object.keys(objSum).forEach((el,idx) => arrSum.push([el, Object.values(objSum)[idx]]));
    
    arrSum.sort((a,b) => b[1] - a[1]);
    
    arrSum.forEach((el,idx) => {
        // [[0,34],[5,324]....]
        obj[el[0]]
        
        for(let i = 0; i < 2; i++){
            if(obj[el[0]][i]){
                result.push(obj[el[0]][i][0]);    
            }
        }
                               
    })
    
    return result;
}

File: 프로그래머스/3/42579. 베스트앨범/베스트앨범.py
def solution(genres, plays):
    answer = []
    play_data = {}
    data = {}
    
    for idx, (genre, play) in enumerate(zip(genres, plays)):
        if genre in data:
            data[genre].append((idx, play))
        else:
            data[genre] = [(idx, play)]
        
        play_data[genre] = play_data.get(genre, 0) + play

    for (genre, _) in sorted(play_data.items(), key=lambda x: x[1], reverse=True):
        for (idx, _) in sorted(data[genre], key=lambda x: x[1], reverse=True)[:2]:
            answer.append(idx)
    
    return answer